pkg/controllers/metrics/nodepool/controller.go

```diff
@@ -95,6 +100,9 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
+	if !nodepoolutils.IsManaged(nodePool, c.cloudProvider) {
```

Comment:
musing: It would be nice if we could generalize this concept somehow outside of the context of the Reconcile loop -- particularly because we are having to repeat ourselves in a bunch of places

Reply:
Discussed offline, will look into a wrapper around the typed reconciler - not sure if it makes sense yet, but could be helpful. Going to decouple though.